core/core.go

```diff
@@ -250,6 +257,42 @@ func initPrivateKey(path string) (p2pcrypto.PrivKey, error) {
 	return nil, err
 }
 
+func initNetworkId(networkID int, db *meshdb.MeshDB) error {
+	// see if there is already a stored networkID
+	snapshot, err := db.Metadata.GetSnapshot()
+	if err != nil {
+		return err
+	}
+
+	filter := db.Metadata.EthereumNetworkIDIndex.All()
```

Comment:
We don't need to use an index, filter, or query. Those features are for efficiently selecting a specific model or group of models from a collection. In this case, we know that the collection only holds exactly zero or one models. We can just get that model directly:

```go
metadata, err := db.Metadata.FindByID([]byte{})
if _, ok := err.(NotFoundError); ok {
    // No previous metadata was found. We need to save the metadata.
}  else if err != nil {
    // Some unexpected error occurred. We should return it.
}
// If we reached here, metadata was previously saved. We should check if the Ethereum
// network id matches what we expect and might also do other checks in the future.
```

Let me know if that makes sense or if you have any other questions.

Reply:
Sweet -- this is much better. Are you passing an empty byte slice to `FindByID` just as an example, or is that exactly how it should be done? 

The key on the collection is `[]byte("metadata")` so should that be used?